### PR TITLE
rgw: datalog list support --shard-id and --marker

### PR DESCRIPTION
**Pull request by tianshan**
Signed-off-by: Tianshan Qu <tianshan@xsky.com>

### CODE REVIEW

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-03-06-2056](https://trello.com/c/gkIitMRd/272-wip-yuri3-testing-2018-03-06-2056)

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -6913,11 +6913,15 @@ int main(int argc, const char **argv)
...
-    RGWDataChangesLog::LogMarker marker;
+    RGWDataChangesLog::LogMarker log_marker;
 
     do {
       list<rgw_data_change_log_entry> entries;
-      ret = log->list_entries(start_time.to_real_time(), end_time.to_real_time(), max_entries - count, entries, marker, &truncated);
+      if (specified_shard_id) {
+        ret = log->list_entries(shard_id, start_time.to_real_time(), end_time.to_real_time(), max_entries - count, entries, marker, NULL, &truncated);
+      } else {
+        ret = log->list_entries(start_time.to_real_time(), end_time.to_real_time(), max_entries - count, entries, log_marker, &truncated);
```
Comment (fangyuxiangGL) [inline src/rgw/rgw_admin.cc]: how about do it only with --marker specified?

Comment (tianshan) [reply src/rgw/rgw_admin.cc]: each datalog has own marker sequence, don't think only --marker is meaningful

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-03-07-1524](https://trello.com/c/gkIitMRd/272-wip-yuri3-testing-2018-03-07-1524)

**Pull request merged**